paimon-core/src/main/java/org/apache/paimon/metrics/MetricGroup.java

```diff
@@ -0,0 +1,141 @@
+     */
+    default Counter counter(int name) {
+        return counter(String.valueOf(name));
+    }
+
+    /**
+     * Creates and registers a new {@link org.apache.paimon.metrics.Counter} with Paimon.
+     *
+     * @param name name of the counter
+     * @return the created counter
+     */
+    Counter counter(String name);
+
+    /**
+     * Registers a {@link org.apache.paimon.metrics.Counter} with Paimon.
+     *
+     * @param name name of the counter
+     * @param counter counter to register
+     * @param <C> counter type
+     * @return the given counter
+     */
+    default <C extends Counter> C counter(int name, C counter) {
+        return counter(String.valueOf(name), counter);
+    }
+
+    /**
+     * Registers a {@link org.apache.paimon.metrics.Counter} with Paimon.
+     *
+     * @param name name of the counter
+     * @param counter counter to register
+     * @param <C> counter type
+     * @return the given counter
+     */
+    <C extends Counter> C counter(String name, C counter);
+
+    /**
+     * Registers a new {@link org.apache.paimon.metrics.Gauge} with Paimon.
+     *
+     * @param name name of the gauge
+     * @param gauge gauge to register
+     * @param <T> return type of the gauge
+     * @return the given gauge
+     */
+    default <T, G extends Gauge<T>> G gauge(int name, G gauge) {
+        return gauge(String.valueOf(name), gauge);
+    }
+
+    /**
+     * Registers a new {@link org.apache.paimon.metrics.Gauge} with Paimon.
+     *
+     * @param name name of the gauge
+     * @param gauge gauge to register
+     * @param <T> return type of the gauge
+     * @return the given gauge
+     */
+    <T, G extends Gauge<T>> G gauge(String name, G gauge);
+
+    /**
+     * Registers a new {@link Histogram} with Paimon.
+     *
+     * @param name name of the histogram
+     * @param histogram histogram to register
+     * @param <H> histogram type
+     * @return the registered histogram
+     */
+    <H extends Histogram> H histogram(String name, H histogram);
+
+    /**
+     * Registers a new {@link Histogram} with Paimon.
+     *
+     * @param name name of the histogram
+     * @param histogram histogram to register
+     * @param <H> histogram type
+     * @return the registered histogram
+     */
+    default <H extends Histogram> H histogram(int name, H histogram) {
+        return histogram(String.valueOf(name), histogram);
+    }
+
+    /**
+     * Gets the scope as an array of the scope components, for example {@code ["myTable",
+     * "bucket-1"]}.
+     */
+    String[] getScopeComponents();
+
+    /**
+     * Returns the fully qualified metric name, for example {@code "myTable.bucket-1.metricName"}.
+     *
+     * @param metricName metric name
+     * @return fully qualified metric name
+     */
+    String getMetricIdentifier(String metricName, char delimiter);
+
+    /** Returns a map of all variables and their associated value. */
+    Map<String, String> getAllVariables();
```

Comment:
What the difference between `scope` and `variable` (or `tag`)? It seems that with scope, the same metric for different buckets have different names. This makes aggregating the metrics difficult.